x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/jwt/JwtRealmSettings.java

```diff
@@ -163,6 +195,12 @@ private static Set<Setting.AffixSetting<SecureString>> getSecureSettings() {
         return new HashSet<>(List.of(HMAC_JWKSET, HMAC_KEY, CLIENT_AUTHENTICATION_SHARED_SECRET));
     }
 
+    public static final Setting.AffixSetting<TokenType> TOKEN_TYPE = Setting.affixKeySetting(
+        RealmSettings.realmSettingPrefix(TYPE),
+        "token_type",
+        key -> new Setting<>(key, TokenType.ID_TOKEN.value(), value -> TokenType.parse(value, key), Setting.Property.NodeScope)
```

Comment:
> Setting.Property.NodeScope

Should this be allowed to be dynamic ? I see the rest are not, but does this make sense just for this setting (maybe part of the PR) or does it make sense along with other settings currently non-dynamic (maybe a follow up PR) ?

Reply:
Currently realm settings are all node scoped since by desgin realms are node specific. We had very brief discussion previously about making realms dynamically configurable. But it is not a small effort. So this work (JWT realm) is likely not the place for tackling it.